### PR TITLE
3242 masquerade tab UI

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Fixes part of #3242 .

Add any further explanations here.

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [x] Is the TravisCI build passing?
* [ ] Was the CHANGELOG.md updated to reflect the changes?
* [x] Was the documentation framework updated to reflect the changes?
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Do all unit tests pass?
* [x] Do all end-to-end tests pass?
* [x] Any other testing performed?
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [x] If applicable, add screenshots or log transcripts of the feature working

image

### CODE REVIEW

File: monkey/monkey_island/cc/ui/src/services/configuration/masquerade.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  'title': 'Masquerade',
+  'type': 'object',
+  'properties': {
+    'masque_list': {
+      'title': 'Masque list',
+      'type': 'array',
+      'uniqueItems': true,
+      'items': {'type': 'string'},
+      'default': [],
+      'description': 'List of masque strings that will be appended to the agent binaries '
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/ui/src/services/configuration/masquerade.js:11]: Don't say "appended". Anywhere. This is an implementation detail. Neither the user, the UI, or the API care at all about the specific implementation details and how this data is added to the binary.

File: monkey/monkey_island/cc/ui/src/services/configuration/masquerade.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+const MASQUERADE = {
+  'title': 'Masquerade',
+  'type': 'object',
+  'properties': {
+    'masque_list': {
+      'title': 'Masque list',
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/ui/src/services/configuration/masquerade.js:6]:
```suggestion
      'title': 'Strings',
```

File: monkey/monkey_island/cc/ui/src/components/pages/ConfigurePage.js [hunk 3]
```diff
@@ -140,8 +144,13 @@ class ConfigurePageComponent extends AuthComponent {
       });
   }
 
+  updateMasqueList = () => {
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/ui/src/components/pages/ConfigurePage.js:147]:
```suggestion
  updateMasqueStrings = () => {
```

File: monkey/monkey_island/cc/ui/src/components/pages/ConfigurePage.js [hunk 4]
```diff
@@ -50,6 +52,7 @@ class ConfigurePageComponent extends AuthComponent {
     this.state = {
       configuration: {},
       credentials: {},
+      masqueList: {},
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/ui/src/components/pages/ConfigurePage.js:55]:
```suggestion
      masqueStrings: {},
```

**Review by mssalvatore — CHANGES_REQUESTED** [review]

File: monkey/monkey_island/cc/ui/src/services/configuration/masquerade.js (hunk 1 above)
Comment (shreyamalviya) [inline monkey/monkey_island/cc/ui/src/services/configuration/masquerade.js:11]: I think we could be more descriptive about what this feature does and why it's added.

File: monkey/monkey_island/cc/ui/src/services/configuration/masquerade.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+const MASQUERADE = {
+  'title': 'Masquerade',
+  'type': 'object',
+  'description': 'To identify malware in a file, one common approach is to search for specific ' +
+  'strings or unique byte sequences.\nIn order to enhance Infection Monkey\'s malware emulation,'+
+  'custom data can be injected into agent binaries, allowing the agents to imitate various types of malware. '+
+  'This feature can be particularly helpful when testing custom detection rules.',
+  'properties': {
+    'masque_linux_strings': {
```
Comment (shreyamalviya) [inline monkey/monkey_island/cc/ui/src/services/configuration/masquerade.js:9]:
```suggestion
    'linux_masque_strings': {
```
For Windows as well

**Review by shreyamalviya — CHANGES_REQUESTED** [review]

File: monkey/monkey_island/cc/ui/src/services/configuration/masquerade.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
+const MASQUERADE = {
+  'title': 'Masquerade',
+  'type': 'object',
+  'description': 'To identify malware in a file, one common approach is to search for specific ' +
+  'strings or unique byte sequences.\nIn order to enhance Infection Monkey\'s malware emulation,'+
+  'custom data can be injected into agent binaries, allowing the agents to imitate various types of malware. '+
+  'This feature can be particularly helpful when testing custom detection rules.',
```
Comment (shreyamalviya) [inline monkey/monkey_island/cc/ui/src/services/configuration/masquerade.js:7]: "Infection Monkey can emulate malware by injecting custom data into Agent binaries. This feature can be particularly helpful when testing custom detection rules."

The rest should be in the documentation, where features are explained better.

File: monkey/monkey_island/cc/ui/src/services/configuration/masquerade.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  'properties': {
+    'linux_masque_strings': {
+      'title': 'Linux Masque Strings',
+      'type': 'array',
+      'uniqueItems': true,
+      'items': {'type': 'string'},
+      'default': [],
+      'description': 'List of masque strings that will be insert to the Linux Agent binaries '
```
Comment (shreyamalviya) [inline monkey/monkey_island/cc/ui/src/services/configuration/masquerade.js:15]:
```suggestion
      'description': 'List of masque strings that will be applied to the Linux Agent binary'
```
We've used "apply" everywhere in the backend. Let's stick to one.
Same for Windows.

File: monkey/monkey_island/cc/ui/src/services/configuration/masquerade.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+  'title': 'Masquerade',
+  'type': 'object',
+  'description': 'Infection Monkey can emulate malware by injecting custom data into Agent binaries.\n'+
+    'This feature can be particularly helpful when testing custom detection rules.',
+  'properties': {
+    'linux_masque_string': {
+      'title': 'Linux Masque',
+      'type': 'string',
+      'default': '',
+      'description': 'A masque string that will be applied to the Linux Agent binaries.'
```
Comment (shreyamalviya) [inline monkey/monkey_island/cc/ui/src/services/configuration/masquerade.js:11]:
```suggestion
      'description': 'A masque string that will be applied to the Linux Agent binary.'
```

**Review by shreyamalviya — APPROVED** [review]
LGTM, but we should talk about whether we want to change the backend to accept a list of masques or put that off right now, before merging.

**Comment by ilija-lazoroski** [comment]
After a discussion, we decided to move on to what the endpoints return which is a single masque. If needed, the change for a list of masques in the UI won't affect the whole feature implementation.

**Pull request merged**